WordPress/gutenberg
# [Query-Total Block] Add editable text and visibility toggle for total count in Query Total block

Enhancements for: #67629

## What?
This update enhances the Query Total block by introducing two key features:

1. Editable text fields for user-defined customization of displayed content.
2. A toggle option for hiding or showing the total count when using the `rangeDisplay` type.

## Why?

- Editable Text provides flexibility for users to modify the default text to match their design or communication needs without overriding block logic.
- Visibility Toggle allows a cleaner design by giving control over whether the total count is displayed alongside the range.

## How?

1. Editable Text: 

    - Added support for rich text editing in the block output for both `total-results` and `range-display` types.

2. Toggle Option:

    - Introduced a new block attribute showTotal.
    - Integrated a toggle control in the block control for visibility management.
    - Adjusted the rangeDisplay rendering logic to include or omit the total count based on user selection.

## Testing Instructions

1. Add a Query Total block to a post or page.
2. Select the block and verify that the displayed text is editable directly in the editor for both the variations.
3. Choose the `range display` option from the toolbar.
4. Toggle the visibility of the total count using the newly added control from the Block controls.
5. Verify that the visibility toggle affects the rendered output in both the editor and frontend.

## Screenshots or screencast 

https://github.com/user-attachments/assets/2ff91a68-3acf-4a03-aa55-bba1a91c2c8b
